### PR TITLE
Tags

### PR DESCRIPTION
**Pull request by ianstanton**
* Allows users to define AWS Tags in their `*.tfvars` file.
* Users define tags as follows:
```
tags = { "key-1" = "value-1",
         "key-2" = "value-2",
         "key-n" = "value-n" 
}
```
* `Deployment ID` tag will be defined by default
* Format changes were made after running `terraform fmt`
* Related to #4

### CODE REVIEW

**Review by kaxil — APPROVED** [review]

**Comment by kaxil** [comment]
I have approved as it looks good but there are Merge conflicts that you will have to resolve @ianstanton

File: bastion.tf [hunk 1]
```diff
@@ -12,26 +12,27 @@ data "aws_ami" "ubuntu" {
   }
 
   owners = ["099720109477"] # Canonical
+  tags   = local.tags
```
Comment (sjmiller609) [inline bastion.tf]: I think we don't need this line

File: db.tf [hunk 2]
```diff
@@ -35,6 +35,8 @@ module "aurora" {
   username            = "airflow"
   password            = local.postgres_airflow_password
   publicly_accessible = false
+
+  tags = local.tags
 }
 
 resource "aws_db_parameter_group" "aurora_db_postgres_parameter_group" {
```
Comment (sjmiller609) [inline db.tf:42]: tags on this resource and the one below (noted in list above)

File: domain.tf [hunk 3]
```diff
@@ -6,7 +6,7 @@ and AWS Route 53
 
 resource "tls_private_key" "lets_encrypt_private_key" {
   algorithm = "RSA"
-  rsa_bits = "4096"
+  rsa_bits  = "4096"
```
Comment (sjmiller609) [inline domain.tf]: I think this is the bit with the merge conflict. we were having issues specifying the key length, so when you resolve the conflicts, please omit the key length in the resource and the rest in this file.

**Review by sjmiller609 — CHANGES_REQUESTED** [review]
It's looking good, and thanks for the formatting. There are a few resources that could also use tags:

- resource "aws_lambda_function" "elb_lookup"
- resource "aws_iam_role" "elb_lookup_role"
- resource "aws_security_group" "all_worker_mgmt"
- Module “vpc”
- resource "aws_key_pair" "bastion_ssh_key"
- resource "aws_db_parameter_group" "aurora_db_postgres_parameter_group"
- resource "aws_rds_cluster_parameter_group" "aurora_cluster_postgres_parameter_group"

**Comment by ianstanton** [comment]
> It's looking good, and thanks for the formatting. There are a few resources that could also use tags:
> 
> * resource "aws_lambda_function" "elb_lookup"
> * resource "aws_iam_role" "elb_lookup_role"
> * resource "aws_security_group" "all_worker_mgmt"
> * Module “vpc”
> * resource "aws_key_pair" "bastion_ssh_key"
> * resource "aws_db_parameter_group" "aurora_db_postgres_parameter_group"
> * resource "aws_rds_cluster_parameter_group" "aurora_cluster_postgres_parameter_group"

@sjmiller609  Added to these resources. Going to do a quick test.

**Comment by ianstanton** [comment]
> > It's looking good, and thanks for the formatting. There are a few resources that could also use tags:
> > 
> > * resource "aws_lambda_function" "elb_lookup"
> > * resource "aws_iam_role" "elb_lookup_role"
> > * resource "aws_security_group" "all_worker_mgmt"
> > * Module “vpc”
> > * resource "aws_key_pair" "bastion_ssh_key"
> > * resource "aws_db_parameter_group" "aurora_db_postgres_parameter_group"
> > * resource "aws_rds_cluster_parameter_group" "aurora_cluster_postgres_parameter_group"
> 
> @sjmiller609 Added to these resources. Going to do a quick test.

Edit: Removed tag from resource "aws_key_pair" "bastion_ssh_key". Key pair does not support tags.

**Review by sjmiller609 — APPROVED** [review]
perfect!

**Comment by sjmiller609** [comment]
https://github.com/astronomer/terraform-aws-astronomer-aws/releases/tag/1.0.9

**Pull request merged**